react/features/settings/components/web/audio/AudioSettingsContent.tsx

```diff
@@ -296,55 +292,82 @@ class AudioSettingsContent extends Component<Props, State> {
+                </ContextMenuItemGroup>
+                { outputDevices.length > 0 && (
+                    <ContextMenuItemGroup>
+                        <ContextMenuItem
+                            accessibilityLabel = { t('settings.speakers') }
+                            className = 'audio-preview-header'
+                            icon = { IconVolumeUp }
+                            id = { this.speakerHeaderId }
+                            text = { t('settings.speakers') } />
                         <ul
-                            aria-labelledby = 'microphone_settings_header'
+                            aria-labelledby = { this.speakerHeaderId }
                             className = 'audio-preview-content-ul'
                             role = 'radiogroup'
-                            tabIndex = '-1'>
-                            {this.state.audioTracks.map((data, i) =>
-                                this._renderMicrophoneEntry(data, i, this.state.audioTracks.length, t)
+                            tabIndex = { -1 }>
+                            { outputDevices.map((data, i) =>
+                                this._renderSpeakerEntry(data, i, outputDevices.length, t)
                             )}
                         </ul>
-                    </div>
-                    { outputDevices.length > 0 && (
-                        <div role = 'menuitem'>
-                            <hr className = 'audio-preview-hr' />
-                            <AudioSettingsHeader
-                                IconComponent = { IconVolumeUp }
-                                id = { this.speakerHeaderId }
-                                text = { t('settings.speakers') } />
-                            <ul
-                                aria-labelledby = 'speaker_settings_header'
-                                className = 'audio-preview-content-ul'
-                                role = 'radiogroup'
-                                tabIndex = '-1'>
-                                { outputDevices.map((data, i) =>
-                                    this._renderSpeakerEntry(data, i, outputDevices.length, t)
-                                )}
-                            </ul>
-                        </div>)
-                    }
-                </div>
-            </div>
+                    </ContextMenuItemGroup>)
+                }
+                {!prejoinVisible && (
+                    <ContextMenuItemGroup>
+                        <div
+                            className = 'audio-preview-checkbox-container'
+                            // eslint-disable-next-line react/jsx-no-bind
+                            onClick = { e => e.stopPropagation() }>
+                            <Checkbox
+                                checked = { noiseSuppressionEnabled }
+                                label = { t('toolbar.noiseSuppression') }
+                                onChange = { toggleSuppression } />
```

Comment:
maybe we should allow the toggle to occur when user clicks the whole row rather than just the checkbox?

Reply:
I don't know. I think it would be better but the last time I discussed this with Tibi he said checkboxes should only be toggled by clicking the actual checkbox not the label